source/geod24/LocalRest.d

```diff
@@ -1176,3 +1228,80 @@ unittest
     node.foo();
     node.bar(1);
 }
+
+// request timeouts (from main thread)
+unittest
+{
+    import core.thread;
+    import std.exception;
+
+    static interface API
+    {
+        size_t sleepFor (long dur);
+    }
+
+    static class Node : API
+    {
+        override size_t sleepFor (long dur)
+        {
+            Thread.sleep(msecs(dur));
+            return 42;
+        }
+    }
+
+    // node with no timeout
+    auto node = RemoteAPI!API.spawn!Node();
+    assert(node.sleepFor(80) == 42);  // no timeout
+
+    // node with a configured timeout
+    auto to_node = RemoteAPI!API.spawn!Node(50.msecs);
+
+    /// none of these should time out
+    assert(to_node.sleepFor(10) == 42);
+    assert(to_node.sleepFor(20) == 42);
+    assert(to_node.sleepFor(30) == 42);
+    assert(to_node.sleepFor(40) == 42);
+
+    auto exc = collectException!Exception(to_node.sleepFor(100));
+    assert(exc !is null);
+}
+
+// request timeouts (foreign node to another node)
+unittest
+{
+    import core.thread;
+    import std.exception;
+
+    static interface API
+    {
+        bool checkTimedOut ();
+        size_t sleepFor (long dur);
+    }
+
+    __gshared API api;  // can't pass it through ctor
```

Comment:
And you don't want to. If you want to contact an API, you store the `Tid` and instantiate a new `RemoteNode`.
`grep` for `tbn` to see how it's done.

Reply:
Wait I fixed this.. where did it go.